### PR TITLE
data/systemd, wrappers: tweak system-shutdown helper for core18

### PR DESCRIPTION
**Pull request by chipaca**
We'd broken the system-shutdown helper service by including its
/bin/sh in the list of things that got redirected to the snapd snap
(which does not ship a /bin/sh). This fixes that, by only rewriting
things that point to /usr/lib/snapd in the first place.

While there I also changed snapd.system-shutdown.service to not call
/bin/sh, but instead use multiple ExecStart= lines. This might make
debug logs from systemd a little bit clearer when things break again.

### CODE REVIEW

File: data/systemd/snapd.system-shutdown.service.in [hunk 1]
```diff
@@ -10,7 +10,9 @@ ConditionPathExists=@libexecdir@/snapd/system-shutdown
 
 [Service]
 Type=oneshot
-ExecStart=/bin/sh -euc 'mount /run -o remount,exec; mkdir -p /run/initramfs; cp @libexecdir@/snapd/system-shutdown /run/initramfs/shutdown'
```
Comment (zyga) [inline data/systemd/snapd.system-shutdown.service.in]: Really nice improvement!

File: wrappers/core18_test.go [hunk 2]
```diff
@@ -69,6 +73,12 @@ func (s *servicesTestSuite) TestAddSnapServicesForSnapdOnCore(c *C) {
 	// and paths get re-written
 	c.Check(string(content), Equals, fmt.Sprintf("[Unit]\nExecStart=%s/snapd/1/usr/lib/snapd/snapd\n# X-Snapd-Snap: do-not-start", dirs.SnapMountDir))
 
+	// check that snapd.system-shutdown.service is created
+	content, err = ioutil.ReadFile(filepath.Join(dirs.SnapServicesDir, "snapd.system-shutdown.service"))
+	c.Assert(err, IsNil)
+	// and paths *do not* get re-written
```
Comment (zyga) [inline wrappers/core18_test.go]: Ah, nice touch to include this!

File: wrappers/core18_test.go [hunk 3]
```diff
@@ -95,6 +105,7 @@ WantedBy=snapd.service
 		{"start", "usr-lib-snapd.mount"},
 		{"daemon-reload"},
 		{"--root", dirs.GlobalRootDir, "enable", "snapd.service"},
+		{"--root", dirs.GlobalRootDir, "enable", "snapd.system-shutdown.service"},
```
Comment (zyga) [inline wrappers/core18_test.go]: Where is this coming from?

Comment (chipaca) [reply wrappers/core18_test.go]: this is used (or used to be used) for image creation, with services shipping enabled. I'm not sure if it's  still used.

**Pull request merged**